test/e2e/upgrades/configmaps.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrades
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	"k8s.io/apimachinery/pkg/util/uuid"
+)
+
+// ConfigMapUpgradeTest tests that a ConfigMap is available before and after
+// a cluster upgrade.
+type ConfigMapUpgradeTest struct {
+	configMap *v1.ConfigMap
+}
+
+// Setup creates a ConfigMap and then verifies that a pod can consume it.
+func (t *ConfigMapUpgradeTest) Setup(f *framework.Framework) {
+	configMapName := "upgrade-configmap"
+
+	// Grab a unique namespace so we don't collide.
+	ns, err := f.CreateNamespace("configmap-upgrade", nil)
+	framework.ExpectNoError(err)
+
+	t.configMap = &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: ns.Name,
+			Name:      configMapName,
+		},
+		Data: map[string]string{
+			"data": "some configmap data",
+		},
+	}
+
+	By("Creating a ConfigMap")
+	if t.configMap, err = f.ClientSet.Core().ConfigMaps(ns.Name).Create(t.configMap); err != nil {
+		framework.Failf("unable to create test ConfigMap %s: %v", t.configMap.Name, err)
+	}
+
+	By("Making sure the ConfigMap is consumable")
+	t.testPod(f)
```

Comment:
Do we want to keep a pod running during the upgrade and make sure the mounted files don't disappear during the upgrade?